src/main/scala/com/sky/kafka/configurator/Main.scala

```diff
@@ -28,24 +27,30 @@ object Main extends LazyLogging {
       .text("Session and connection timeout for Zookeeper")
   }
 
-  def main(args: Array[String]): Unit = run(args) match {
-    case Success(_) =>
-      System.exit(0)
-    case Failure(_) =>
-      System.exit(1)
-  }
-
-  def run(args: Array[String]): Try[Unit] = parse(args) flatMap { conf =>
+  def main(args: Array[String]): Unit = {
     logger.info(s"Running ${BuildInfo.name} ${BuildInfo.version}")
-    val configurator = TopicConfigurator.reader(conf)
-    val result: Try[Unit] = for {
-      topics <- TopicConfigurationParser(new FileReader(conf.file))
-      _ <- topics.map(configurator.configure).sequenceU
-    } yield ()
-    stopApp(configurator)
-    result
+    run(args) match {
+      case Success((logs, _)) =>
+        logs.foreach(log => logger.info(log))
+        System.exit(0)
+      case Failure(t) =>
+        logger.error(s"Failed to configure topics", t)
```

Comment:
I've not run this myself, but here's my concern:  If we partially succeed in configuring a few topics, then in the old system we'd get all those logs out first, and then get a failure message when something went wrong.

It *looks* to me like in that case, we'd now only get the failure message, and never know which parts had succeeded prior to the failure.

Am I wrong?